src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,236 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * This class handles the logic behind the Duke chatbot
+ */
+public class Duke {
+
+    private Ui ui;
+    private Storage storage;
+    private InputParser inputParser;
+
+    public Duke() {
+        this.ui = new Ui();
+        this.storage = new Storage();
+        this.inputParser = new InputParser();
+    }
+
+    /**
+     * Runs the Duke chat bot.
+     *
+     * @throws DukeException
+     */
+    public void run() throws DukeException {
```

Comment:
This method is very long, almost 200 lines of code. Should try to create other classes/methods to reduce the length of the method.

Reply:
I will refactor the code into different classes once I am done with the GUI. Thank you for this.